libs/sdk-ui-vis-commons/src/legend/PopUpLegend/PopUpLegend.tsx

```diff
@@ -21,21 +21,33 @@ export interface IPopUpLegendProps {
     enableBorderRadius?: boolean | ItemBorderRadiusPredicate;
     containerId: string;
 
+    onPageChanged?: (page: number) => void;
     customComponent?: JSX.Element | null;
 }
 
 /**
  * @internal
  */
 export const PopUpLegend: React.FC<IPopUpLegendProps> = (props) => {
-    const { name, maxRows, enableBorderRadius, series, onLegendItemClick, containerId, customComponent } =
-        props;
+    const {
+        name,
+        maxRows,
+        enableBorderRadius,
+        series,
+        onLegendItemClick,
+        containerId,
+        customComponent,
+        onPageChanged = () => {},
+    } = props;
     const intl = useIntl();
     const [isDialogOpen, setDialogOpen] = useState(false);
 
     const dialogTitle = name || intl.formatMessage({ id: "properties.legend.title" });
 
-    const onCloseDialog = () => setDialogOpen(false);
+    const onCloseDialog = () => {
+        setDialogOpen(false);
+        onPageChanged(1);
```

Comment:
Could you please explain to me why is needed to use this `onPageChanged`? 🙂

Reply:
Well, I think it's good to reset the paging when you close the dialog, so that, when reopened, it opens on the very first page.

Reply:
And it was the case that if you closed the dialog on page 5, it reopened on page 5?

Reply:
Actually it opened on the very first page, but the parent component which renders the title did not know about it and still rendered title for page 5 after reopening the dialog.